core/src/main/java/com/linecorp/armeria/server/ServiceNotFoundException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+import com.linecorp.armeria.common.util.Exceptions;
+
+/**
+ * A {@link RuntimeException} raised when a {@link Service} failed to find a resource.
+ */
+public final class ServiceNotFoundException extends RuntimeException {
```

Comment:
Ok, understood your point. I didn't want to call `DynamicHttpFunctionEntry.bind` twice because it invoked pattern matching and path-variable extraction. But basically I agree your suggestion. I will think about that more to make the codes efficient.  :)